src/wrapper/clap-juce-wrapper.cpp

```diff
@@ -1000,7 +1006,55 @@ class ClapJuceWrapper : public clap::helpers::Plugin<
         if (!editor->isResizable())
             return false;
 
-        editor->setSize(static_cast<int>(*w), static_cast<int>(*h));
+        auto cst = editor->getConstrainer();
+
+        if (!cst)
+            return true; // we have no constraints. Whaever is fine!
+
+        auto minW = (uint32_t)cst->getMinimumWidth();
+        auto maxW = (uint32_t)cst->getMaximumWidth();
+        auto minH = (uint32_t)cst->getMinimumHeight();
+        auto maxH = (uint32_t)cst->getMaximumHeight();
+
+        // There is no std::clamp in c++14
+        auto width = *w;
+        if (width < minW)
+            width = minW;
+        if (width > maxW)
+            width = maxW;
+
+        auto height = *h;
+        if (height < minH)
+            height = minH;
+        if (height > maxH)
+            height = maxH;
+
+        auto aspectRatio = (float)cst->getFixedAspectRatio();
+
+        if (aspectRatio != 0.0)
+        {
+            // This is an unsatisfactory but stable algorithm.
+            // Since we don't know drag direction it is hard to
+            // find a stable better one, at least it is for me right
+            // now
+            width = aspectRatio * height;
```

Comment:
Just to make sure I'm understanding: the problem here is that we don't know if the user is trying change the width or the height, so we're assuming they're trying to change the height, and setting the width accordingly?

If that's the case, what we could do:
- check if the requested width and height are different from the current width and height
- if the height has changed more than the width, set `width = aspectRatio * height;`
- if the width has changed more than the height, set `height = (1.0 / aspectRatio) * width;`

That said, I worry that clamping the width and height above could make it weird for detecting which direction to resize in.

Reply:
Right so the Problrm is if you do thst a very small motion plus int rounding make things unstable. I tried a bunch of algos including that one and this was the only one which didn’t throw a misbehaving exception. 

If you just add the algo you outlined un comment the terminate at the top and use it in bitwig mac it terminates almost immediately on resize 

but also that algo doesn’t work when shrinking width it seems. So if you then try to capture a last size you run into state problems 

and then alas if you update last size and try and keep it on a timer so you can get it consistent through a drag it gets really hairy!!!